### PR TITLE
Added Command as a replacement for Actions

### PR DESCRIPTION
**Pull request by RomanTysiachnik**
## Command - replacement of Actions (closures)
This new type was added to replace our old implementation of closures usage. 
CommandWith is a generic struct with type T, where T is a closure that you send.
To present your variable as a Command, you should specify its type. By default, Command has an empty closure that doesn't receives and doesn't returns anything:`(Void) -> (Void)`. To use any other type as a received one, you should specify it as `CommandWith<Type>`.

### CODE REVIEW

File: PlayerControls/sources/ContentControlsViewController.swift [hunk 1]
```diff
@@ -147,7 +149,7 @@ extension ContentControlsViewController.Props.Player.Item {
             public struct Option {
                 public var name = ""
                 public var selected = false
-                public var select: Action<Void> = nop
+                public var select: Command? = Command.nop
```
Comment (AlexeyDemedetskiy) [inline PlayerControls/sources/ContentControlsViewController.swift]: nil may be?

File: PlayerControls/sources/ControlsVisibilityController.swift [hunk 2]
```diff
@@ -26,52 +26,51 @@ public class ControlsPresentationController {
...
+    public func play() { behavior.perform(with: .play) }
+    public func pause() { behavior.perform(with: .pause) }
+    public func resetTimer() { behavior.perform(with: .resetTimer) }
     
     private enum Message {
         case tap, timerFired, play, pause, resetTimer
     }
     
-    private var behavior: Action<Message> = { fatalError("Unhandled message: \($0)") }
+    private var behavior: CommandWith<Message> = CommandWith { fatalError("Unhandled message: \($0)") }
```
Comment (AlexeyDemedetskiy) [inline PlayerControls/sources/ControlsVisibilityController.swift]: Can we use `.fatalError("message")`?

File: PlayerControls/sources/Timer.swift [hunk 3]
```diff
@@ -4,9 +4,9 @@ import Foundation
 
 public class Timer {
     private var timer: Foundation.Timer!
-    private let fire: Action<Void>
-    public init(duration: TimeInterval, fire: @escaping Action<Void>) {
-        self.fire = fire
+    private let fire: Command
+    public init(duration: TimeInterval, fire: Command?) {
+        self.fire = fire!
```
Comment (AlexeyDemedetskiy) [inline PlayerControls/sources/Timer.swift]: WTF?

File: PlayerControls/sources/Command.swift [hunk 4]
```diff
@@ -0,0 +1,32 @@
+//  Copyright © 2017 One by AOL : Publishers. All rights reserved.
+
+public typealias Command = CommandWith<Void>
+
+public struct CommandWith<T> {
+    private var action: (T) -> Void
```
Comment (BogdanBilonog) [inline PlayerControls/sources/Command.swift]: If use `perform` instead `action` struct will become more elegant

Comment (RomanTysiachnik) [reply PlayerControls/sources/Command.swift]: Command with action -> Command with perform
Don't you think that with action it looks much more readable?

Comment (BogdanBilonog) [reply PlayerControls/sources/Command.swift]: If `action` will be renamed with `perform` and not `private`, then `initi` and `perform` methods can be removed too

Comment (RomanTysiachnik) [reply PlayerControls/sources/Command.swift]: Well, it wouldn't change anything. Are there any other suggestions?

**Comment by AndriiMoskvin** [comment]
Discussion summary: rename `xxxCommand` to `xxxAction` in variable names.

**Comment by AndriiMoskvin** [comment]
@RomanTysiachnik If you haven't started this yet - we discussed this with @AlexeyDemedetskiy - we can follow `onXxx` pattern here. This will hide the implementation details (`Action` or `Command`) - and still will be clear named.

**Comment by RomanTysiachnik** [comment]
@AndriiMoskvin @AlexeyDemedetskiy `onXxx`? Maybe better `xxxOn`?

**Comment by RomanTysiachnik** [comment]
@AndriiMoskvin! I've discussed with @AlexeyDemedetskiy about these names and I'm agreed with him, that it should be left as it was (with Actions). It looks and sounds perfect and if we change these names, it won't change anything eventually. If there would be an urgent need to change it, I will do it in another PR.

**Pull request merged**